helix-core/src/main/java/org/apache/helix/controller/stages/MessageGenerationPhase.java

```diff
@@ -70,6 +70,8 @@
       .getSystemPropertyAsLong(SystemPropertyKeys.CONTROLLER_MESSAGE_PURGE_DELAY, 60 * 1000);
   private final static String PENDING_MESSAGE = "pending message";
   private final static String STALE_MESSAGE = "stale message";
+  // TODO: Make the message retry count configurable through the Cluster Config or IdealStates.
+  public final static int DEFAULT_STATE_TRANSITION_MESSAGE_RETRY_COUNT = 3;
```

Comment:
Is this the same thing as the message attribute of retrying? Otherwise, we need an accurate name since it is the retry for state model creation.

Reply:
It is the same. So in theory it will be applied to the timeout condition too. Although ST message does not have a timeout setup for now, we may have it in the future. So make this name too explicit may be confusing as well. I propose to make it a general retry count for ST messages. But it is debatable. Let me know what's your thought.